### PR TITLE
Deprecates "funding account" and account creation options

### PR DESCRIPTION
**Pull request by KPrasch**
The name says it all.

### CODE REVIEW

File: nucypher/blockchain/eth/actors.py [hunk 1]
```diff
@@ -512,8 +515,8 @@ def initialize_stake(self,
         if entire_balance:
             amount = self.token_balance
         if not self.token_balance >= amount:
-            raise self.StakerError(f"Insufficient token balance ({self.token_agent}) "
-                                   f"for new stake initialization of {amount}")
+            raise self.InsufficientTokens(f"Insufficient token balance ({self.token_agent}) "
```
Comment (tuxxy) [inline nucypher/blockchain/eth/actors.py]: :+1:

File: nucypher/blockchain/eth/actors.py [hunk 2]
```diff
@@ -988,63 +924,24 @@ def set_worker(self, staker_address: str, worker_address: str, password: str = N
...
+                         checksum_address: str,
                          password: str = None,
-                         fund_now: bool = True
                          ) -> Stake:
 
         # Existing Staker address
-        if checksum_address:
-            if not is_checksum_address(checksum_address):
-                raise ValueError(f"{checksum_address} is an invalid EIP-55 checksum address.")
+        if not is_checksum_address(checksum_address):
```
Comment (tuxxy) [inline nucypher/blockchain/eth/actors.py]: Don't we have a decorator for this, IIRC?

File: nucypher/cli/stake.py [hunk 3]
```diff
@@ -30,8 +30,6 @@
 @click.option('--poa', help="Inject POA middleware", is_flag=True)
 @click.option('--offline', help="Operate in offline mode", is_flag=True)
 @click.option('--provider-uri', help="Blockchain provider's URI i.e. 'file:///path/to/geth.ipc'", type=click.STRING)
-@click.option('--funding-address', help="Address to stake NU ERC20 tokens", type=EIP55_CHECKSUM_ADDRESS)
-@click.option('--pre-funded', help="Do not fund new stake's accounts", is_flag=True, default=False)
```
Comment (tuxxy) [inline nucypher/cli/stake.py]: :+1:

File: tests/blockchain/eth/entities/actors/test_stakeholder.py [hunk 4]
```diff
@@ -50,68 +50,25 @@ def test_software_stakeholder_configuration(testerchain,
...
-    assert staker.stakes[0] == stake
-
-    # Ensure stake is on-chain via agency
-    staking_agent = Agency.get_agent(StakingEscrowAgent)
-    stakes = list(staking_agent.get_all_stakes(staker_address=stake.owner_address))
-    assert len(stakes) == 1
-
-
-def test_initialize_stake_with_existing_staking_account(software_stakeholder, stake_value, token_economics):
+def test_initialize_stake_with_existing_account(software_stakeholder, stake_value, token_economics):
```
Comment (tuxxy) [inline tests/blockchain/eth/entities/actors/test_stakeholder.py]: :+1:

File: tests/blockchain/eth/entities/actors/test_stakeholder.py [hunk 5]
```diff
@@ -50,68 +50,25 @@ def test_software_stakeholder_configuration(testerchain,
...
+    with pytest.raises(IndexError):
+        stake = software_stakeholder.stakes[0]
 
-    # Really... there is one stake.
+    # Really... there are no stakes.
     staking_agent = Agency.get_agent(StakingEscrowAgent)
-    stakes = list(staking_agent.get_all_stakes(staker_address=stake.owner_address))
-    assert len(stakes) == 1
+    stakes = list(staking_agent.get_all_stakes(staker_address=software_stakeholder.accounts[0]))
+    assert len(stakes) == 0
```
Comment (tuxxy) [inline tests/blockchain/eth/entities/actors/test_stakeholder.py]: :+1:

File: tests/blockchain/eth/entities/actors/test_stakeholder.py [hunk 6]
```diff
@@ -120,20 +77,19 @@ def test_initialize_stake_with_existing_staking_account(software_stakeholder, st
 
     # Ensure the stakeholder is tracking the new staker and stake.
     assert len(software_stakeholder.stakers) == 1
-    assert len(software_stakeholder.stakes) == 2
+    assert len(software_stakeholder.stakes) == 1
```
Comment (tuxxy) [inline tests/blockchain/eth/entities/actors/test_stakeholder.py]: :+1:

File: tests/fixtures.py [hunk 7]
```diff
@@ -689,6 +687,12 @@ def manual_staker(testerchain):
 
     txhash = testerchain.client.w3.eth.sendTransaction(tx)
     _receipt = testerchain.wait_for_receipt(txhash)
+
+    token_agent = Agency.get_agent(NucypherTokenAgent)
+    token_agent.transfer(amount=NU(200_000, 'NU').to_nunits(),
```
Comment (tuxxy) [inline tests/fixtures.py]: :+1:

File: tests/blockchain/eth/entities/actors/test_stakeholder.py [hunk 8]
```diff
@@ -50,68 +50,25 @@ def test_software_stakeholder_configuration(testerchain,
...
-    assert len(stakes) == 1
-
-
-def test_initialize_stake_with_existing_staking_account(software_stakeholder, stake_value, token_economics):
+def test_initialize_stake_with_existing_account(software_stakeholder, stake_value, token_economics):
 
     # There is one staker and one stake.
-    assert len(software_stakeholder.stakers) == 1
-    assert len(software_stakeholder.stakes) == 1
+    assert len(software_stakeholder.stakers) == 0
```
Comment (michwill) [inline tests/blockchain/eth/entities/actors/test_stakeholder.py]: Is the comment above still valid?

**Comment by michwill** [comment]
Looks good, but somehow I have the following problem.
If I run all tests with py.test, it fails with `eth.exceptions.OutOfGas: Out of gas: Needed 4424000 - Remaining 4375218 - Reason: Write contract code for CREATE`. But somehow, if I run the failing test with `py.test -x tests/crypto/test_powers.py`, it works fine.

**Comment by michwill** [comment]
Despite having the registry set, `nucypher stake` still tries to use memory registry in this PR:
```
nucypher stake --registry-filepath ~/.local/share/nucypher/dev_contract_registry.json list
```
raises an exception with the following message:
```
nucypher.blockchain.eth.registry.NoRegistry: No registry at filepath: ::memory-registry::
```

**Comment by KPrasch** [comment]
@michwill - I suspect that you were running the tests at the same time you were running this commend?  I think there is an outstanding bug, wherein your config file is overwritten by some test.  Looking into this (This is Issue is not introduced in this PR).

File: tests/blockchain/eth/entities/actors/test_stakeholder.py (hunk 8 above)
Comment (KPrasch) [reply tests/blockchain/eth/entities/actors/test_stakeholder.py]: No.

**Comment by michwill** [comment]
@KPrasch not at the same time: before. But I see what you say - maybe can remove all configs after running tests

**Comment by KPrasch** [comment]
@michwill - I added an additional mock to help ensure that the default configuration root is not being written to the disk during tests. I do not know for certain that it is fixed.

File: tests/blockchain/eth/entities/actors/test_stakeholder.py (hunk 8 above)
Comment (KPrasch) [reply tests/blockchain/eth/entities/actors/test_stakeholder.py]: :heavy_check_mark:

**Pull request merged**